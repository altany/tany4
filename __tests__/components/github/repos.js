import { render } from '@testing-library/react'
import Repos from '../../../components/github/repos'

jest.mock('swr')
const swr = require('swr').default

it('renders repos without language unchanged', () => {
    const Repos = require('../../../components/github/repos').default
    const { container } = render(<Repos />)
    expect(container).toMatchSnapshot()
})

it('renders repos for language unchanged', () => {
    const data=['repo1', 'repo 2', 'repo 3']
    const error=undefined
    swr.mockImplementation(()=>({data, error}))
    
    const { container } = render(<Repos language='Javascript' />)
    expect(container).toMatchSnapshot()
})

it('renders repos for language and no data unchanged', () => {
    const data=undefined
    const error=undefined
    swr.mockImplementation(()=>({data, error}))

    const { container } = render(<Repos language='Javascript' />)
    expect(container).toMatchSnapshot()
})

it('renders repos for language with error unchanged', () => {
    const data=undefined
    const error='test error'
    swr.mockImplementation(()=>({data, error}))

    const { container } = render(<Repos language='Javascript' />)
    expect(container).toMatchSnapshot()
})