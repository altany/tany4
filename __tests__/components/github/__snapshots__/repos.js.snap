// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a repo with language and error unchanged 1`] = `<div />`;

exports[`renders a repo with language and no data unchanged 1`] = `<div />`;

exports[`renders a repo with language unchanged 1`] = `
<div>
  <h4>
    Repos with 
    <span>
      Javascript
    </span>
  </h4>
  <ul>
    <li>
      <a
        href="https://github.com/altany/repo1"
        target="_blank"
      >
        repo1
      </a>
    </li>
    <li>
      <a
        href="https://github.com/altany/repo 2"
        target="_blank"
      >
        repo 2
      </a>
    </li>
    <li>
      <a
        href="https://github.com/altany/repo 3"
        target="_blank"
      >
        repo 3
      </a>
    </li>
  </ul>
</div>
`;

exports[`renders a repo without language unchanged 1`] = `<div />`;
